src/Orchard.Cms.Web/Modules/Orchard.Roles/Services/RoleStore.cs

```diff
@@ -58,10 +58,10 @@ public void UpdateRoles(RolesDocument roles)
             _memoryCache.Set(Key, roles);
         }
 
-        public async Task<IEnumerable<string>> GetRoleNamesAsync()
+        public async Task<IEnumerable<RoleNamesEntry>> GetRoleNamesAsync()
```

Comment:
Why not return the normalized name only then, and if you need the displayed one then you just look it up using the service?